### PR TITLE
Added Swerve Lock Method and used in execute

### PR DESCRIPTION
**Pull request by outsidermm**

### CODE REVIEW

File: components/chassis.py [hunk 1]
```diff
@@ -294,6 +312,14 @@ def on_enable(self) -> None:
     def get_imu_speed(self) -> float:
         return math.hypot(self.imu.getVelocityX(), self.imu.getVelocityY())
 
+    def lock_swerve(self) -> None:
+        if not self.swerve_lock:
```
Comment (OliverW10) [inline components/chassis.py:316]: These checks are redundant

File: components/chassis.py [hunk 2]
```diff
@@ -261,6 +262,20 @@ def drive_local(self, vx: float, vy: float, omega: float) -> None:
     def execute(self) -> None:
         # rotate desired velocity to compensate for skew caused by discretization
         # see https://www.chiefdelphi.com/t/field-relative-swervedrive-drift-even-with-simulated-perfect-modules/413892/
+        if self.swerve_lock:
+            # Actuating the swerve to face to the center of the robot, not movable
+            desired_states = (
+                SwerveModuleState(0, Rotation2d(math.radians(135))),
```
Comment (auscompgeek) [inline components/chassis.py:268]: use Rotation2d.fromDegrees

File: components/chassis.py [hunk 3]
```diff
@@ -276,11 +275,27 @@ def execute(self) -> None:
...
+        else:
+            desired_states = self.kinematics.toSwerveModuleStates(desired_speeds)
+            desired_states = self.kinematics.desaturateWheelSpeeds(
+                desired_states, attainableMaxSpeed=self.max_wheel_speed
+            )
+
         for state, module in zip(desired_states, self.modules):
+            if abs(state.speed) < 0.01 and not self.swerve_lock:
+                module.drive.set(ctre.ControlMode.Velocity, 0)
+                module.steer.set(ctre.ControlMode.PercentOutput, 0)
```
Comment (auscompgeek) [inline components/chassis.py:297]: This logic really doesn't belong here. The modules should know how to handle this themselves.

Can we add a lock method to the modules instead?

**Review by OliverW10 — APPROVED** [review]
LGTM after robot test

**Pull request merged**